src/platform/nrfconnect/wifi/WiFiManager.h

```diff
@@ -140,31 +145,31 @@ class WiFiManager
     CHIP_ERROR Connect(const ByteSpan & ssid, const ByteSpan & credentials, const ConnectionHandling & handling);
     StationStatus GetStationStatus() const;
     CHIP_ERROR ClearStationProvisioningData();
-    CHIP_ERROR DisconnectStation();
+    CHIP_ERROR Disconnect();
     CHIP_ERROR GetWiFiInfo(WiFiInfo & info) const;
     CHIP_ERROR GetNetworkStatistics(NetworkStatistics & stats) const;
 
 private:
-    CHIP_ERROR AddPsk(const ByteSpan & credentials);
+    CHIP_ERROR AddPsk(wifi_connect_req_params * params, const ByteSpan & credentials);
     CHIP_ERROR EnableStation(bool enable);
     CHIP_ERROR AddNetwork(const ByteSpan & ssid, const ByteSpan & credentials);
-    void PollTimerCallback();
-    void WaitForConnectionAsync();
-    void OnConnectionSuccess();
-    void OnConnectionFailed();
     uint8_t GetSecurityType() const;
 
-    WpaNetwork * mpWpaNetwork{ nullptr };
-    ConnectionCallback mConnectionSuccessClbk;
-    ConnectionCallback mConnectionFailedClbk;
-    System::Clock::Timeout mConnectionTimeoutMs;
-    ScanCallback mScanCallback{ nullptr };
+    // Event handling
+    static void WifiMgmtEventHandler(struct net_mgmt_event_callback * cb, uint32_t mgmtEvent, struct net_if * iface);
+    static void mScanResultClbk(struct net_mgmt_event_callback * cb);
+    static void mScanDoneClbk(struct net_mgmt_event_callback * cb);
+    static void mConnectClbk(struct net_mgmt_event_callback * cb);
+    static void mDisconnectClbk(struct net_mgmt_event_callback * cb);
 
     static uint8_t FrequencyToChannel(uint16_t freq);
-    static StationStatus StatusFromWpaStatus(const wpa_states & status);
 
-    static const Map<wpa_states, StationStatus, 10> sStatusMap;
-    static const Map<uint16_t, uint8_t, 42> sFreqChannelMap;
+    struct wifi_connect_req_params mWiFiParams = {};
+    static uint32_t mScanResult;
```

Comment:
I think it shouldn't be static. If you want to access it from the static methods, you can create static method to get WiFiManager object instance and access this field that way.

Reply:
The `WiFiManager` is a singleton anyway. So Instance().mScanResult should do the job with mScanResult being non-static class member. BTW. if static, the `m` prefix does not apply here.